### PR TITLE
JSDK-2670 Handle Chrome and Firefox iOS user agents.

### PR DESCRIPTION
**Pull request by manjeshbhargav**
@makarandp0 

This PR has the following changes:

* Make sure Chrome and Firefox are detected correctly on iOS.
* Consolidate browser version detection in guessBrowserVersion().

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: lib/util/index.js [hunk 1]
```diff
@@ -104,25 +104,68 @@ function flatMap(list, mapFn) {
...
-    if (/Chrome/.test(navigator.userAgent)) {
-      return 'chrome';
-    }
-    if (/Firefox/.test(navigator.userAgent)) {
-      return 'firefox';
-    }
-    if (/Safari/.test(navigator.userAgent)) {
-      return 'safari';
-    }
+function guessBrowser(userAgent) {
```
Comment (makarandp0) [inline lib/util/index.js]: I see that you are using `userAgent` parameter only from unit test. consider changing: 
`function guessBrowser(userAgent)` =>  `function guessBrowser(userAgent = getUserAgent())`

File: test/unit/spec/util/index.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+var assert = require('assert');
+var util = require('../../../../lib/util');
+
+describe('Util', () => {
```
Comment (makarandp0) [inline test/unit/spec/util/index.js]: nice tests!

File: lib/getstats.js [hunk 3]
```diff
@@ -2,16 +2,16 @@
 
 var flatMap = require('./util').flatMap;
 var guessBrowser = require('./util').guessBrowser;
+var guessBrowserVersion = require('./util').guessBrowserVersion;
 var getSdpFormat = require('./util/sdp').getSdpFormat;
 
 var guess = guessBrowser();
+var guessVersion = guessBrowserVersion();
```
Comment (makarandp0) [inline lib/getstats.js]: nit: guessVersion => browserVersion

File: lib/getstats.js [hunk 4]
```diff
@@ -2,16 +2,16 @@
 
 var flatMap = require('./util').flatMap;
 var guessBrowser = require('./util').guessBrowser;
+var guessBrowserVersion = require('./util').guessBrowserVersion;
 var getSdpFormat = require('./util/sdp').getSdpFormat;
 
 var guess = guessBrowser();
```
Comment (makarandp0) [inline lib/getstats.js]: nit: guess => guessedBrowser

File: lib/util/index.js [hunk 5]
```diff
@@ -104,25 +104,68 @@ function flatMap(list, mapFn) {
...
+ */
+function guessBrowserVersion(userAgent) {
+  if (typeof userAgent === 'undefined') {
+    userAgent = getUserAgent();
+  }
+  var prefix = {
+    chrome: 'Chrome|CriOS',
+    firefox: 'Firefox|FxiOS',
+    safari: 'Version'
+  }[guessBrowser(userAgent)];
```
Comment (makarandp0) [inline lib/util/index.js]: This looks bit dangerous as as `guessBrowser` could return null.

File: lib/util/index.js (hunk 1 above)
Comment (manjeshbhargav) [reply lib/util/index.js]: @makarandp0 This project is still `es5`, so we can do this once we move to `.ts`.

File: lib/util/index.js (hunk 5 above)
Comment (manjeshbhargav) [reply lib/util/index.js]: @makarandp0 `null` results in `prefix` having the value of `undefined`, which is handled in the next line.

**Pull request merged**